src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7656,15 +7656,22 @@ public Encoding Encoding
 
             set
             {
-                // Check for UTF-7 by checking for code page 65000
-                // See: https://docs.microsoft.com/en-us/dotnet/core/compatibility/corefx#utf-7-code-paths-are-obsolete
-                if (value != null && value.CodePage == 65000)
-                {
-                    _provider.WriteWarning(PathUtilsStrings.Utf7EncodingObsolete);
-                }
+                WarnIfUtf7Encoding(value);
                 _encoding = value;
                 // If an encoding was explicitly set, be sure to capture that.
                 WasStreamTypeSpecified = true;
+
+                void WarnIfUtf7Encoding(Encoding value)
+                {
+                    const int CodePageUtf7 = 65000;
+
+                    // Use the recommended pattern to check for UTF-7.
+                    // https://docs.microsoft.com/dotnet/core/compatibility/core-libraries/5.0/utf-7-code-paths-obsolete
+                    if (value is { CodePage: CodePageUtf7 })
+                    {
+                        _provider.WriteWarning(PathUtilsStrings.Utf7EncodingObsolete);
+                    }
+                }
```

Comment:
This seems overkill to me. I prefer the original code.

Reply:
@xtqqczze Can you please respond to this comment? Thanks!

Reply:
I prefer the symbolic constant instead of a magic number.

https://refactoring.guru/replace-magic-number-with-symbolic-constant

Reply:
@iSazonov, @SteveL-MSFT Thoughts?

Reply:
I guess @daxian-dbw don't like the modern syntax but ok with the new name `WarnIfUtf7Encoding`. If so I am with you too.

Reply:
I should be clearer on my comment.
It's not the syntax `value is { CodePage: CodePageUtf7 }`, but that you have a local method in a setter. Please don't use the local method.

Reply:
@xtqqczze - please respond to this feedback